spirv_hlsl.cpp

```diff
@@ -1508,7 +1512,45 @@ void CompilerHLSL::emit_buffer_block(const SPIRVariable &var)
 
 void CompilerHLSL::emit_push_constant_block(const SPIRVariable &var)
 {
-	emit_buffer_block(var);
+	std::cout << options.root_constants_layout.size() << std::endl;
+	if (options.root_constants_layout.empty())
+	{
+		emit_buffer_block(var);
+		return;
+	}
+	else
+	{
+		for(auto const& layout : options.root_constants_layout)
+		{
+			auto &type = get<SPIRType>(var.basetype);
+
+			flattened_structs.insert(var.self);
+			type.member_name_cache.clear();
+			add_resource_name(var.self);
+			auto &memb = meta[type.self].members;
+
+			statement("cbuffer ", to_name(var.self), ": register(b", layout.start / 4, ", space0)");
```

Comment:
Is this binding assignment arbitrary? It probably makes sense to add explicit namespace/binding for the RootConstant struct?

Reply:
It might be better to add some kind of begin/end range support to the already implemented emit_buffer_block() functions to reduce maintenance.

Reply:
> Is this binding assignment arbitrary? It probably makes sense to add explicit namespace/binding for the RootConstant struct?

right, that's quite tailored towards our specific root signature layout (space0, binding equals to start of range). I'm going to make it a bit more generic.